### PR TITLE
Color correction

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Adds a level of color correction to station areas. It's not overly complex or probably true color correction, but it works for our purposes. This also includes mapping helpers, one of these can be placed in an area to change that entire area's correction.

Essentially it's a subtle color-shift to areas

## Why It's Good For The Game

Emphasizes the visual atmosphere of a given area

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

(null)scrnshot1

New Project (1)

New Project (2)

New Project (3)

New Project (4)

New Project

</details>

## Changelog
:cl:
add: Adds color correction to areas
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
The transition effect needs to be extremely slow and the colours need to be very, very subtle to prevent this being too noticeable

**Comment by DrDuckedGoose** [comment]
This is probably fine for now

File: code/game/sound.dm [hunk 1]
```diff
@@ -255,7 +255,7 @@ distance_multiplier - Can be used to multiply the distance at which the sound is
 	UNTIL(SSticker.login_music) //wait for SSticker init to set the login music
 
 	if(prefs && (prefs.toggles & PREFTOGGLE_SOUND_LOBBY))
-		SEND_SOUND(src, sound(SSticker.login_music, repeat = 0, wait = 0, volume = vol, channel = CHANNEL_LOBBYMUSIC)) // MAD JAMS
+		SEND_SOUND(src, sound(SSticker.login_music, repeat = 0, wait = 0, volume = 0, channel = CHANNEL_LOBBYMUSIC)) // MAD JAMS
```
Comment (Penwin0) [inline code/game/sound.dm]: ?

File: code/modules/mob/living/living_movement.dm [hunk 2]
```diff
@@ -2,6 +2,9 @@
 	. = ..()
 	update_turf_movespeed(loc)
 	update_looking_move()
+	//color correction
+	var/area/A = get_area(loc)
+	apply_color_correction(A?.color_correction)
```
Comment (Penwin0) [inline code/modules/mob/living/living_movement.dm]: A bit worried about this potentially getting expensive

Comment (Penwin0) [reply code/modules/mob/living/living_movement.dm]: Could employ COMSIG_MOVABLE_ENTERED_AREA instead?

File: code/game/sound.dm (hunk 1 above)
Comment (DrDuckedGoose) [reply code/game/sound.dm]: :trollface:

File: code/modules/mob/living/living_movement.dm (hunk 2 above)
Comment (DrDuckedGoose) [reply code/modules/mob/living/living_movement.dm]: That's heaps better, yeah

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -1435,3 +1437,9 @@
 			layer = initial(layer)
 		if(.) //We weren't pone before, so we become dense and things can bump into us again.
 			density = initial(density)
+
+//Used for applying color correction
+/mob/living/proc/apply_color_correction(datum/source, area/entered)
```
Comment (Penwin0) [inline code/modules/mob/living/living.dm:1445]: Missing SIGNAL_HANDLER

**Review by Penwin0 — DISMISSED** [review]

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (DrDuckedGoose) [reply code/modules/mob/living/living.dm:1445]: 👍

**Review by PowerfulBacon — COMMENTED** [review]
The fade needs to be way longer, at least 30 seconds.
Only warm/dark colours should be used for colour fading. Purples and greens don't work with colour correction very well.

**Comment by DrDuckedGoose** [comment]
increased fade time & changed hydro & science to cold_ish

**Comment by PowerfulBacon** [comment]
Putting this on testmerge again

**Review by Penwin0 — APPROVED** [review]
My code reviews were adressed, and I think it looks good in game

**Comment by DrDuckedGoose** [comment]
> Will users have an option to disable this, as they do with ambient occlusion?

No, but if someone hates it that much, they can surely ask someone else(not me) to do that

**Comment by CydiaLamiales** [comment]
> Will users have an option to disable this, as they do with ambient occlusion?

Game not being fullbright burns the eyes of the /vg/ player

**Comment by PowerfulBacon** [comment]
I would rather not bloat preference anymore with unnecessary options that players don't know the effect of.

**Comment by DrDuckedGoose** [comment]
Do you not understand the concept of unpaid work?

**Comment by CydiaLamiales** [comment]
> > > Will users have an option to disable this, as they do with ambient occlusion?
> > 
> > 
> > No, but if someone hates it that much, they can surely ask someone else(not me) to do that
> 
> Why do you, the person adding the feature, expect someone _else_ to add a remedy if you end up creating a problem?

There is no problem. Only you hate that the game doesn't look like it's 2014.

**Comment by PowerfulBacon** [comment]
I still don't want to add too many settings. I don't want players to have the option to make their game look worse and if this looks worse than without it and is too obvious, I won't be merging it.

**Comment by PowerfulBacon** [comment]
> Point Sampling vs Bilinear Filtering vs Nearest Neighbor, or for ambient occlusion, or to enable and disable tint from glasses?

Ambient occlusion could genuinely be removed, tint from glasses just gets in the way so kind of needs to be turned off and while I would love to remove the setting to change scaling mode, different monitor resolutions look blurry on certain settings so you need to choose the one for you

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**